src/components/TodoList.js

```diff
@@ -0,0 +1,46 @@
+import { useTodoStorage } from "../useTodoStorage";
+import TodoForm from "./TodoForm";
+import TodoItem from "./TodoItem";
+
+function TodoList() {
+    const [todos, setTodos] = useTodoStorage("todos");
+
+    function onSubmit(todo) {
+        if (todo === "") {
+            return;
+        }
+        setTodos([
+            ...todos,
+            { id: new Date().valueOf(), name: todo, isComplete: false },
+        ]);
+    }
+
+    function toggleTodo(id) {
+        const updateTodos = todos.map((todo) =>
+            todo.id === id ? { ...todo, isComplete: !todo.isComplete } : todo
+        );
+        setTodos([...updateTodos]);
```

Comment:
(optional) Based on your logic, you can write this code like this.

setTodos((prevTodos) => [...prevTodos.map((todo) =>
            todo.id === id ? { ...todo, isComplete: !todo.isComplete } : todo
        )])

it's better to access previous state that is passed from setState.